src/http.ts

```diff
@@ -11,60 +11,30 @@ import { GenericError } from './errors';
 
 export const createAbortController = () => new AbortController();
 
-const switchFetch = async (
+const fetchAndSerialize = async (
```

Comment:
Do we need this function now or could we inline it into `fetchWithoutWorker`? It's just I'm not sure the name is ideal and wonder if we're hitting a function fatigue of sorts. I would have liked to have just called it `getJSON`, but we've already got one of those!

Reply:
I introduced this function because we need to use it in promise.race 

```
return Promise.race([
    fetchAndSerialize(fetchUrl, fetchOptions),
    new Promise((_, reject) => {
      timeoutId = setTimeout(() => {
        controller.abort();
        reject(new Error("Timeout when executing 'fetch'"));
      }, timeout);
    })
  ]).finally(() => {
    clearTimeout(timeoutId);
  });
```
As it was using await, followed by a mapping of the response, I wasnt able to use Promise.race without moving it in a function.

Next I decided to only define the function ones instead of recreating the same function every time the switchFetch gets called, but I can inline that inside the switchFetch if u want. But it will still have to stay a function.

Reply:
My original thinking was it could just be something like this (untested):

```js
return Promise.race([
  fetch(fetchUrl, fetchOptions).then(async(response) => ({
      ok: response.ok,
      json: await response.json()
    }),
  new Promise((_, reject) => {
      timeoutId = setTimeout(() => {
        controller.abort();
        reject(new Error("Timeout when executing 'fetch'"));
      }, timeout);
    })
]);
```

Without the overhead of another function and the burden of having to name it. It just seems like we're wrapping a simple fetch call without too much benefit.

Reply:
I can give that a try!

Reply:
Nah, it doesn't :D let's go with what you have and apologies for being a blocker

Reply:
Well, it does but you have to cast it:

```js
return Promise.race([
    fetch(fetchUrl, fetchOptions).then(async response => {
      return {
        ok: response.ok,
        json: await response.json()
      } as any;
    }),
    new Promise((_, reject) => {
      timeoutId = setTimeout(() => {
        controller.abort();
        reject(new Error("Timeout when executing 'fetch'"));
      }, timeout);
    })
  ]).finally(() => {
    clearTimeout(timeoutId);
  });
```

Reply:
I am not sure about using that. Sure it avoids a function, but it adds a cast to any that I dont realy understant why.

Reply:
That's fair. Ultimately though I don't think the `fetchAndSerialize` name works though (for example, if anything we're deserializing, not serializing). Maybe we can just call it `doFetch`?